src/AssetManager.h

```diff
@@ -6,18 +6,31 @@
 
 #include "GL/glew.h"
 #include "SDL_opengl.h"
+#include "SDL.h"
 
 #include "MeshAsset.h"
+#include "TextureAsset.h"
+#include "ShaderAsset.h"
 
 class AssetManager {
 public:
     AssetManager();
     ~AssetManager();
 
-    MeshAsset* get_mesh_asset(const std::string& filepath);
+    MeshAsset* get_mesh_asset(const std::string& file_path);
+    TextureAsset* get_texture_asset(const std::string& file_path);
+    ShaderAsset* get_shader_asset(const std::string& file_path);
```

Comment:
Returning a reference would be better I think. For our case, if we couldn't load something (i.e. this return value doesn't exist), let's just crash the program. I can't imagine a case for us that would not be useful.